esrally/driver/runner.py

```diff
@@ -230,6 +229,20 @@ def _transport_request_params(params):
         return params, request_params, transport_params, headers
 
 
+class Configurator:
+    """
+    Mixin to create configuration-aware runners
+    """
+
+    def __init__(self, config, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.serverless_mode = False
+        self.serverless_operator = False
+        if config:
+            self.serverless_mode = config.opts("driver", "serverless.mode", mandatory=False, default_value=False)
+            self.serverless_operator = config.opts("driver", "serverless.operator", mandatory=False, default_value=False)
```

Comment:
I haven't used multiple inheritance much before as I've always been told to avoid it at all costs. I don't know how we all feel about this, but presumably it wouldn't hurt to have all runners be 'configuration' aware, in which case we could just add this logic to `Runner` itself.. I think, I'm probably overlooking something obvious.

Reply:
I had `config` in `Runner` initially but had not found the right way to express it's optional character, and then got inspired by `Delegator` mixin. I went back to `Runner` in https://github.com/elastic/rally/pull/1789/commits/670de571987c35866d00737cbe647cacb889cb0b. PTAL.